src/screens/PodcastsScreen.tsx

```diff
@@ -41,81 +47,142 @@ export class PodcastsScreen extends React.Component<Props, State> {
       } else {
         const userToken = await RNSecureKeyStore.get('BEARER_TOKEN')
         if (userToken) {
-          getAuthUserInfo()
+          await getAuthUserInfo()
         }
       }
     } catch (error) {
       console.log(error.message)
     }
+
+    this.setState({ isLoading: false })
   }
 
-  selectLeftItem = (fromSelected: string) => {
-    this.setState({ fromSelected })
+  selectLeftItem = async (selectedKey: string) => {
+    if (!selectedKey) {
+      this.setState({ queryFrom: null })
+      return
+    }
+
+    const { settings } = this.global
+    const { nsfwMode } = settings
+
+    this.setState({
+      isLoading: true,
+      queryFrom: selectedKey
+    })
+
+    let podcasts = []
+    const newState = {
+      isLoading: false,
+      selectedCategory: ''
+    } as any
+    if (selectedKey === _subscribedKey) {
+      const { session } = this.global
+      const { subscribedPodcastIds } = session.userInfo
+      newState.podcasts = await getSubscribedPodcasts(subscribedPodcastIds || [])
+    } else if (selectedKey === _allPodcastsKey) {
+      const querySort = 'alphabetical'
+      podcasts = await getPodcasts({ sort: querySort }, nsfwMode)
+      newState.podcasts = podcasts[0]
+      newState.querySort = querySort
+    } else if (selectedKey === _categoryKey) {
+      const querySort = 'alphabetical'
+      podcasts = await getPodcasts({
+        categories: selectedKey,
+        sort: querySort
+      }, nsfwMode)
+      newState.querySort = querySort
+      newState.selectedCategory = selectedKey
+    }
+
+    this.setState(newState)
+  }
+
+  selectRightItem = async (selectedKey: string) => {
+    if (!selectedKey) return
+
+    const { settings } = this.global
+    const { nsfwMode } = settings
+    const { queryFrom, selectedCategory } = this.state
+
+    this.setState({
```

Comment:
Seems super odd that there is a setState here and then another one down below in the same function. Would we want to rethink about this a little bit? I think you might want to do the rest of this function inside of setStates completion block maybe

Reply:
Yeah, I am worried about async with two state calls in the same method...but how else do we display a loading spinner to the user before the method completes? It's pretty important that we do...

Reply:
setState takes a second argument. Its a function that runs as soon as setState finishes. You can run everything else inside there. Here is an example:

`this.setState({isLoading:true, querySort: selectedKey}, () => {
    ...do the rest of the function in here. Even another setState
})`